pkg/util/ipset/ipset.go

```diff
@@ -72,74 +72,148 @@ var EntryMemberPattern = "(?m)^(.*\n)*Members:\n"
 // ipset version output is similar to "v6.10".
 var VersionPattern = "v[0-9]+\\.[0-9]+"
 
+// Option is variadic initializer for IPSet
+type Option func(set *IPSet)
+
+// NewIPSet returns a new IPSet struct
+func NewIPSet(name string, setType Type, opts ...Option) (IPSet, error) {
+	ret := IPSet{
+		name:    name,
+		setType: setType,
+	}
+	setIPSetDefaults(&ret)
```

Comment:
You shouldn't need `setIPSetDefaults` any more, because everyone should be using `NewIPSet` to construct their IPSets. Just fill the defaults in above.

Reply:
Hi @danwinship , I am working on it now, do you mean I should define the constructor like following
``` golang
func NewIPSet(name string, setType Type, hashFamily string, hashSize int, maxElem int, portRange string, comment string) IPSet {
	ret := IPSet{
		name:       name,
		setType:    setType,
		hashFamily: hashFamily,
		hashSize:   hashSize,
		maxElem:    maxElem,
		portRange:  portRange,
		comment:    comment,
	}
	return ret
}
```
`You shouldn't need setIPSetDefaults any more` means that this function do not set any default value, the user of this function should know the default values of each field like following in `pkg/proxy/ipvs/ipset.go`, BTW there is only one place reference the function.

```
defaultHashSize, defaultMaxElem := 1024, 65536
set := &IPSet{
IPSet: *utilipset.NewIPSet(
	name, setType, hashFamily,
	defaultHashSize, defaultMaxElem, utilipset.DefaultPortRange,
	comment,
)
```